pkg/common/cluster_state.go

```diff
@@ -18,11 +20,11 @@ import (
 // get from the current state to the desired state
 type DesiredClusterState []ClusterAction
 
-func (d DesiredClusterState) AddAction(action ClusterAction) DesiredClusterState {
+func (d *DesiredClusterState) AddAction(action ClusterAction) DesiredClusterState {
```

Comment:
Are you sure this is correct? Could you please explain why do we need a pointer here? We seem to get it's value for `append`.

Reply:
To allow using pointers to a `DesiredClusterState`. When we declare the method on the non-pointer type then we can't modify the value or pass it to another function. We're limited to pass by value which creates a copy.

Reply:
Ok, thanks for the explanation.